### ISSUE TITLE
Use multiple routes as args 

### ISSUE THREAD

**Issue by nfishe** [fastify/fastify#151]
Would this be a useful feature?  IMO args are prettier than chaining for certain use cases....
```js
fastify.routes(
  {
    method: 'GET',
    url: '/',
    schema: {
      response: {
        '2xx': {
          type: 'object',
          properties: {
            hello: {
              type: 'string'
            }
          }
          }
      }
    },
    handler: function (req, reply) {
      reply.send({ hello: 'world' })
    }
  },
  {
    method: 'GET',
    url: '/moon',
    schema: {
        response: {
        '2xx': {
          type: 'object',
          properties: {
            goodbye: {
              type: 'string'
            }
          }
        }
      }
    },
    handler: function (req, reply) {
      reply.send({ goodbye: 'moon' })
    }
  }
)
```

Maybe with a method like so?
```js
  function routes (args) {
    const options = Array.isArray(args) ? args : [].slice.call(args)

    return options.reduce(function (instance, opts) {
      return instance.route(opts)
    }, this)
  }
```

I am loving this project and its hackability.

**Comment by delvedor** [comment]
We can definitely do that, I love semantic code.
But I don't know if it is better update the actual api and make it work like `register` (that accepts both a single function and an array of functions) instead of adding a new one.
@mcollina thoughts?

> I am loving this project and its hackability.

Thank you :)

**Comment by coopnd** [comment]
Potentially dissect the `route` manager function and let the opts param be an array? I'm guessing only for the route method, leaving the shorthand methods alone....

**Comment by delvedor** [comment]
> Potentially dissect the route manager function and let the opts param be an array?

I didn't get you, can you rephrase?

> I'm guessing only for the route method, leaving the shorthand methods alone....

Yes, only the `route` method :)

**Comment by delvedor** [comment]
Closing this, if you want to continue the discussion feel free to reopen :)

**Comment by mannyluvstacos** [comment]
Hi @delvedor and @mcollina!

Taking from @nfishe's example and modifying it slightly, would it be useful for `route` to take in an Array of objects which are each full declarations of routes?

```javascript
fastify.routes([
  {
    method: 'GET',
    url: '/',
    schema: {
      response: {
        '2xx': {
          type: 'object',
          properties: {
            hello: {
              type: 'string'
            }
          }
          }
      }
    },
    handler: function (req, reply) {
      reply.send({ hello: 'world' })
    }
  },
  {
    method: 'GET',
    url: '/moon',
    schema: {
        response: {
        '2xx': {
          type: 'object',
          properties: {
            goodbye: {
              type: 'string'
            }
          }
        }
      }
    },
    handler: function (req, reply) {
      reply.send({ goodbye: 'moon' })
    }
  }
])
``` 

This would be similar to how Hapi.js handles multiple route registrations:
https://github.com/hapijs/hapi/blob/master/API.md#-serverrouteroute

Currently,  I am registering multiple routes via a `forEach`:
```javascript
routes.forEach(function (entry) {
  fastify.route(entry)
})
```
Where `entry` is a full route declaration.
I look forward to hearing back!

-Manny

**Comment by mcollina** [comment]
Would you like to send a PR?

**Comment by mannyluvstacos** [comment]
I would love to!
I'll start working on it. :)

**Comment by jsumners** [comment]
I'm pretty sure this was a feature that was removed as a "don't need to maintain this code" removal.

**Comment by mcollina** [comment]
Oh really? I don't really remember.

@delvedor?

**Comment by delvedor** [comment]
Yeah, I also remember we decided to drop this :)
You can achieve it easily with:
```js
[routes].forEach(fastify.route.bind(fastify))
```

**Comment by mcollina** [comment]
My bad, sorry @mannyluvstacos.

**Comment by mannyluvstacos** [comment]
:) Not a problem, @mcollina!

Thank you, @jsumners and @delvedor for the context and guidance!
